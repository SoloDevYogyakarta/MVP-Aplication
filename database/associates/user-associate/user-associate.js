"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.userHistoryInclude = exports.userInclude = exports.userAttribute = exports.userAssociate = void 0;
const history_entity_1 = require("../../../database/entities/authenticate/history-entity/history-entity");
const user_entity_1 = require("../../../database/entities/authenticate/user-entity/user-entity");
const order_associate_1 = require("../order-associate/order-associate");
const userAttribute = {
    include: [
        ['createdAt', 'created_at'],
        ['updatedAt', 'updated_at'],
    ],
    exclude: ['password', 'createdAt', 'updatedAt'],
};
exports.userAttribute = userAttribute;
const userInclude = [
    {
        model: history_entity_1.userHistoryEntity,
        attributes: {
            include: [['updatedAt', 'updated_at']],
            exclude: [],
        },
        as: 'activities',
    },
];
exports.userInclude = userInclude;
const userHistoryInclude = [
    {
        model: order_associate_1.orderAssociate,
        attributes: order_associate_1.orderAttribute,
        include: order_associate_1.orderInclude,
        as: 'order',
    },
];
exports.userHistoryInclude = userHistoryInclude;
user_entity_1.userEntity.hasMany(history_entity_1.userHistoryEntity, {
    sourceKey: 'id',
    foreignKey: { name: 'user_id', allowNull: true },
    as: 'activities',
});
user_entity_1.userEntity.hasMany(order_associate_1.orderAssociate, {
    sourceKey: 'id',
    foreignKey: { name: 'user_id', allowNull: true },
    as: 'order',
});
const userAssociate = user_entity_1.userEntity;
exports.userAssociate = userAssociate;
//# sourceMappingURL=user-associate.js.map