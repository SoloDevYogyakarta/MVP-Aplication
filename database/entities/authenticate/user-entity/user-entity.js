"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.userEntity = exports.UserEntity = void 0;
const sequelize_1 = require("sequelize");
const base_entity_1 = require("../../base-entity/base-entity");
const entity_1 = require("../../entity");
class UserEntity extends base_entity_1.BaseEntity {
}
exports.UserEntity = UserEntity;
exports.userEntity = entity_1.sequelize.define('USER', {
    plat_number: sequelize_1.DataTypes.STRING,
    phone_number: sequelize_1.DataTypes.STRING,
    name: sequelize_1.DataTypes.STRING,
    motor: sequelize_1.DataTypes.STRING,
    year_production: sequelize_1.DataTypes.INTEGER,
    address: sequelize_1.DataTypes.STRING,
    role: sequelize_1.DataTypes.STRING,
    password: sequelize_1.DataTypes.STRING,
}, {
    tableName: 'USER',
    schema: 'AUTHENTICATE',
});
//# sourceMappingURL=user-entity.js.map