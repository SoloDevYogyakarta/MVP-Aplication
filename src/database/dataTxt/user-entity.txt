{"id":1088,"public_id":"io7NL8MF7mPHdla16p3X0","username":"updated-Opal_Rempel28","email":"Dameon_Cronin16@gmail.com","phone_number":null,"password":"$2b$10$ix8JcUoVPanbpNiAczvphut0WyjV/wcUUGw0/WpCedOFk.dkqSpvm","createdAt":"2023-10-21T12:41:46.989Z","updatedAt":"2023-10-21T12:41:47.100Z"}