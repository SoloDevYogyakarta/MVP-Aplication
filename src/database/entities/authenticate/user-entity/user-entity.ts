import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { SuperTest } from 'supertest';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class UserEntity extends BaseEntity {
  plat_number: string;
  phone_number: string;
  name: string;
  motor: string;
  year_production: number;
  address: string;
  role: string;
  password: string;
}

export type UserCreationAttribute = Optional<UserEntity, 'id'>;
export interface UserInstance
  extends Model<UserCreationAttribute, UserEntity>,
    UserEntity {}
export const userEntity = sequelize.define<UserInstance>(
  'USER',
  {
    plat_number: DataTypes.STRING,
    phone_number: DataTypes.STRING,
    name: DataTypes.STRING,
    motor: DataTypes.STRING,
    year_production: DataTypes.INTEGER,
    address: DataTypes.STRING,
    role: DataTypes.STRING,
    password: DataTypes.STRING,
  } as ModelAttributes<UserInstance>,
  {
    tableName: 'USER',
    schema: 'AUTHENTICATE',
  },
);
