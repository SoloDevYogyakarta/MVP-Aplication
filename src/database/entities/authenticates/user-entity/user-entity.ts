import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { nanoid } from 'nanoid';
import bcrypt from 'bcrypt';
import { sequelize } from '../../entity';
import { BaseEntity } from '../../base-entity/base-entity';

export class UserEntity extends BaseEntity {
  username: string;
  email: string;
  phone_number: number;
  password: string;
  role: string;
  file_id: string;
}

export type UserCreationAttribute = Optional<UserEntity, 'id'>;
export interface UserInstance
  extends Model<UserCreationAttribute, UserEntity>,
    UserEntity {}
export const userEntity = sequelize.define<UserInstance>(
  'USER',
  {
    public_id: DataTypes.STRING,
    username: DataTypes.STRING,
    email: DataTypes.STRING,
    phone_number: DataTypes.INTEGER,
    role: DataTypes.STRING,
    password: DataTypes.STRING,
    file_id: DataTypes.STRING,
  } as ModelAttributes<UserInstance>,
  {
    tableName: 'USER',
    schema: 'AUTHENTICATE',
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
        instance.password = bcrypt.hashSync(instance.password, 10);
      },
    },
  },
);
