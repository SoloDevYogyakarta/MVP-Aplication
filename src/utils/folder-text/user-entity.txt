{"id":86,"plat_number":"B 3897 SHA","phone_number":"515-480-6929 x04288","name":"Mrs. Elena Hirthe","motor":"Incredible Soft Hat","address":"83045 Vandervort Via","year_production":2084,"role":"admin","password":"$2b$10$dGAlMMGMbtxdE3Zy8vBnnuZ/z96CuhTD0P5Lqudbq2YrqbiDaK2Tm","updatedAt":"2023-11-05T09:14:31.131Z","createdAt":"2023-11-05T09:14:31.131Z","public_id":"tGTvVwVy1_1wTna8BDaqu"}