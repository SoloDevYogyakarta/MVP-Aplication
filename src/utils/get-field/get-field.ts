import { GetFieldType } from 'src/types/get-field.type';
import { readpath } from '../system/system';

export function getField(filename?: GetFieldType) {
  try {
    if (filename === 'fake') {
      return readpath(`../../database/dataTxt/${filename}.txt`);
    }
    return JSON.parse(readpath(`../../database/dataTxt/${filename}.txt`));
  } catch (err) {
    throw new Error(err);
  }
}
