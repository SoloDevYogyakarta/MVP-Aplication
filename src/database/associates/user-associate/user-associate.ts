import { FindAttributeOptions, Includeable } from 'sequelize';
import { userHistoryEntity } from '../../../database/entities/authenticate/history-entity/history-entity';
import { userEntity } from '../../../database/entities/authenticate/user-entity/user-entity';

const userAttribute: FindAttributeOptions = {
  include: [],
  exclude: ['id', 'password'],
};

const userInclude: Includeable[] = [
  {
    model: userHistoryEntity,
    attributes: {
      include: [],
      exclude: ['id', 'user_id'],
    },
    as: 'history',
  },
];

userEntity.hasMany(userHistoryEntity, {
  sourceKey: 'public_id',
  foreignKey: { name: 'user_id', allowNull: true },
  as: 'history',
});

const userAssociate = userEntity;

export { userAssociate, userAttribute, userInclude };
