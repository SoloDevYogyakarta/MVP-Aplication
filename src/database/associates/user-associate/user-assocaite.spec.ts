import { ModelCtor } from 'sequelize';
import { getfield } from '../../../utils/get-field/get-field';
import { UserInstance } from '../../../database/entities/authenticate/user-entity/user-entity';
import { userAssociate, userAttribute, userInclude } from './user-associate';

describe('UserAssociate', () => {
  let public_id!: string;
  let entity: ModelCtor<UserInstance>;

  beforeEach(() => {
    entity = userAssociate;
  });

  it('should to be defined', () => expect(entity).toBeDefined());

  it('render correctly', () => expect(entity).toMatchSnapshot());

  try {
    public_id = getfield('user-http-entity').public_id;
  } catch (err) {
    // empty
  }

  if (public_id) {
    it('findOne with relationship', async () => {
      const findOne = await entity.findOne({
        where: { public_id },
        attributes: userAttribute,
        include: userInclude,
      });
      expect(findOne.public_id).toEqual(public_id);
    });

    it('findAll with relationship', async () => {
      const findAll = await entity.findAll();
      expect(findAll.length).not.toEqual(0);
    });
  }
});
