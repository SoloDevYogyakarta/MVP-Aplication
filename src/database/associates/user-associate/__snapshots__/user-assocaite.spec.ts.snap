// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAssociate render correctly 1`] = `[Function]`;
